### PR TITLE
Fixes #1218 Error: tried to insert existing hash - the two objects were NOT equal

### LINKED ISSUE #1218: Error: tried to insert existing hash - the two objects were NOT equal
Getting the following error when running a plugin:
```
2016-10-18T01:26:55.850Z - error: [gme:server:worker:simpleworker:pid_5051:WorkerFunctions] plugin [ExecutePipeline] failed with error Error: Operation "/K/g/0/b" failed! {"hash":"a68cd90b43e9a3562a431828860d4eaadc56f96d","resultHashes":{"net":"0aa103fbd3b22ab1f5b39f75e6a1ba24c89b0357","stdout":"8bca6cde62795dbc4de1702b8c3eb57e34acb006","train-all-files":"6410d7213a1417a97ca3ee217cbea7d2fc608ef1"},"resultSuperSet":null,"userId":[],"status":"FAILED_TO_EXECUTE","createTime":"2016-10-18T01:26:52.575Z","startTime":"2016-10-18T01:26:54.994Z","finishTime":"2016-10-18T01:26:55.346Z","worker":"mjolnir_5099","labels":[],"outputNumber":0}
    at finish (/home/irishninja/projects/dfst/deepforge/node_modules/webgme/src/server/worker/workerrequests.js:129:56)
    at ._callback (/home/irishninja/projects/dfst/deepforge/node_modules/webgme/src/plugin/managerbase.js:232:21)
    at save.then (/home/irishninja/projects/dfst/deepforge/src/plugins/ExecutePipeline/ExecutePipeline.js:368:30)
    at _fulfilled (/home/irishninja/projects/dfst/deepforge/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/home/irishninja/projects/dfst/deepforge/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/home/irishninja/projects/dfst/deepforge/node_modules/q/q.js:796:13)
    at /home/irishninja/projects/dfst/deepforge/node_modules/q/q.js:604:44
    at runSingle (/home/irishninja/projects/dfst/deepforge/node_modules/q/q.js:137:13)
    at flush (/home/irishninja/projects/dfst/deepforge/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
2016-10-19T01:04:12.438Z - error: [gme:server:standalone:mongo] tried to insert existing hash - the two objects were NOT equal  newObject=["Object","_hasownrelation",true,"_id","#9d6a2246bc23c75ad03067afeac7dd962678eed8","atr",["Object"]], oldObject=["Object","_hasownrelation",true,"_id","#9d6a2246bc23c75ad03067afeac7dd962678eed8"]
2016-10-19T01:04:12.443Z - error: [gme:server:worker:simpleworker:pid_22544:WorkerFunctions:PluginManagerBase:plugin:ExecutePipeline] Error: tried to insert existing hash - the two objects were NOT equal #9d6a2246bc23c75ad03067afeac7dd962678eed8
2016-10-19T01:04:12.440Z - error: [gme:server:standalone:storage] Error: tried to insert existing hash - the two objects were NOT equal #9d6a2246bc23c75ad03067afeac7dd962678eed8
2016-10-19T03:38:15.764Z - error: [gme:server:standalone:mongo] tried to insert existing hash - the two objects were NOT equal  newObject=["Object","_hasownrelation",true,"_id","#9d6a2246bc23c75ad03067afeac7dd962678eed8","atr",["Object"]], oldObject=["Object","_hasownrelation",true,"_id","#9d6a2246bc23c75ad03067afeac7dd962678eed8"]
2016-10-19T03:38:15.769Z - error: [gme:server:standalone:storage] Error: tried to insert existing hash - the two objects were NOT equal #9d6a2246bc23c75ad03067afeac7dd962678eed8
2016-10-19T03:38:15.771Z - error: [gme:server:worker:simpleworker:pid_22762:WorkerFunctions:PluginManagerBase:plugin:ExecutePipeline] Error: tried to insert existing hash - the two objects were NOT equal #9d6a2246bc23c75ad03067afeac7dd962678eed8
```

### PR DESCRIPTION
**Pull request by kecso**
The collision of hashes was the result of a faulty persist operation (that contained empty object(s)).
Now there is no empty object possible as the result of persist, so no faulty 'no-op' json patch will be generated that would lead to this unusual error.

### CODE REVIEW

File: test/perf/core.perf.js [hunk 1]
```diff
@@ -394,13 +389,6 @@ describe.skip('Core Performance test', function () {
                         meta = core.getOwnJsonMeta(allMetaNodes[i]);
                     }
                     console.timeEnd('getOwnJsonMeta');
-
```
Comment (brollb) [inline test/perf/core.perf.js]: Why is this removed?

File: test/common/core/coretype.spec.js [hunk 2]
```diff
@@ -1161,11 +1161,30 @@ describe('coretype', function () {
...
-                    if(--neededChecks === 0){
+                    if (--neededChecks === 0) {
                         done();
                     }
                 }, core.loadInstances(core.getBase(newChild)));
             }, core.loadByPath(newRoot, '/template/child'));
         }, core.loadRoot(core.getHash(root)));
     });
+
+    it('should remove atr and reg field of an instance during persist', function (done) {
```
Comment (brollb) [inline test/common/core/coretype.spec.js]: probably a dumb question but these can be removed, right? Are the attributes and registry values stored in different fields?

File: test/perf/core.perf.js (hunk 1 above)
Comment (kecso) [inline test/perf/core.perf.js]: it is completely unrelated.
we removed the 'meta' user as it was inefficient and didn't gave that much extra functionality.
and though are automated test-suite completely ignores this file, on my local machines it gave me headaches....
so I removed the portion from the performance tests which used the old implementation (there is no alternate on this level, as it was already a function covered by two implementations)

File: test/common/core/coretype.spec.js (hunk 2 above)
Comment (brollb) [inline test/common/core/coretype.spec.js]: That makes sense - it looks like the check on line 180 (checking if the attr/reg is empty) was simply the problem for this issue

File: src/common/core/coretree.js [hunk 3]
```diff
@@ -165,18 +165,19 @@ define([
...
             delete data[CONSTANTS.MUTABLE_PROPERTY];
+            keys = Object.keys(data);
 
             for (i = 0; i < keys.length; i++) {
                 key = keys[i];
                 child = data[key];
                 if (__isMutableData(child)) {
                     sub = __saveData(child, root, path + '/' + key);
-                    if (sub === __getEmptyData()) {
+                    if (JSON.stringify(sub) === JSON.stringify(__getEmptyData())) {
```
Comment (pmeijer) [inline src/common/core/coretree.js]: Isn't it enough to check if subs keys are of length 0?
